### PR TITLE
refactor: bouncer cleanup

### PR DESCRIPTION
**Pull request by j4m1ef0rd**
# Pull Request

Closes: PRO-1382, PRO-785

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [ ] I have updated documentation where appropriate.

## Summary

- Renamed `provideLiquidity` -> `depositLiquidity`
- Changed the new swap command to log the deposit address.
- Fixed the use of the chainflip api in the lp api test.
	- Made the observe swap function error if the api becomes disconnected.
- Added a new function `executeWithTimeout` to use instead of `runWithTimeout` that logs the execution time and handles the process exit.
	- Refactored most commands and tests to use the new timeout function.
	- Changed some timeouts that were too large compared to the actual execution time.

### CODE REVIEW

File: bouncer/shared/utils.ts [hunk 1]
```diff
@@ -344,12 +344,16 @@ export async function observeEvent(
   eventQuery?: EventQuery,
   stopObserveEvent?: () => boolean,
   finalized = false,
+  debug = false,
```
Comment (msgmaxim) [inline bouncer/shared/utils.ts:367]: I know that you added something similar in a few other places already (e.g. `doPerformSwap`), but I wonder if instead of passing parameters for this we should use and enviornment variable instead (similar to RUST_LOG). Then maybe (in a separate PR) we could write a wrapper around console.log to check that env variable. Wouldn't be too difficult to have different levels as well. Alternatively, we could look into using a third party library for this. The way we do it right now is awkward and inconsistent.

File: bouncer/shared/lp_api_test.ts [hunk 2]
```diff
@@ -70,6 +70,8 @@ async function provideLiquidityAndTestAssetBalances() {
 }
 
 async function testRegisterLiquidityRefundAddress() {
+  await using chainflip = await getChainflipApi();
```
Comment (msgmaxim) [inline bouncer/shared/lp_api_test.ts:72]: Why is this necessary when we have `chainflip` in the global scope?

File: bouncer/commands/get_usdc_balance.ts [hunk 3]
```diff
@@ -7,19 +7,12 @@
...
+  console.log(await getErc20Balance('Ethereum', ethereumAddress, contractAddress));
 }
 
-const ethereumAddress = process.argv[2] ?? '0';
-
-runWithTimeout(getUsdcBalanceCommand(ethereumAddress), 5000).catch((error) => {
-  console.error(error);
-  process.exit(-1);
-});
+await executeWithTimeout(getUsdcBalanceCommand(process.argv[2] ?? '0'), 5);
```
Comment (msgmaxim) [inline bouncer/commands/get_usdc_balance.ts:18]: IMO the code was more clear with `const ethereumAddress = process.argv[2] ?? '0';` uninlined (otherwise it took me a while to understand what we are reading from command line arguments). Same in other places.

File: bouncer/shared/utils.ts (hunk 1 above)
Comment (kylezs) [reply bouncer/shared/utils.ts:367]: We could also leverage a lightweight js logging library, which likely does this plus some more potentially useful features

File: bouncer/shared/lp_api_test.ts (hunk 2 above)
Comment (kylezs) [reply bouncer/shared/lp_api_test.ts:72]: When using `await using` I think best not to put in the global scope, as it works like `Drop` in rust. Becomes much harder to reason about when it's in the global scope. So if there is a global usage, we should change that. 

Note that this function is doing something like an Arc.clone() and isn't re-initialising the whole client.

File: bouncer/commands/get_usdc_balance.ts (hunk 3 above)
Comment (martin-chainflip) [reply bouncer/commands/get_usdc_balance.ts:18]: agreed

File: bouncer/shared/utils.ts (hunk 1 above)
Comment (msgmaxim) [reply bouncer/shared/utils.ts:367]: That's what I meant by
> Alternatively, we could look into using a third party library for this.

😉

File: bouncer/shared/lp_api_test.ts (hunk 2 above)
Comment (msgmaxim) [reply bouncer/shared/lp_api_test.ts:72]: Ah, I just realised that the global one was deleted. I was mostly saying that it didn't make sense to keep both. Must have looked at my local branch.

File: bouncer/shared/utils.ts (hunk 1 above)
Comment (kylezs) [reply bouncer/shared/utils.ts:367]: oh lol, missed that completely

Comment (kylezs) [reply bouncer/shared/utils.ts:367]: Created a ticket: https://linear.app/chainflip/issue/PRO-1415/use-a-js-logging-library-in-bouncer

File: bouncer/commands/get_usdc_balance.ts (hunk 3 above)
Comment (msgmaxim) [reply bouncer/commands/get_usdc_balance.ts]: I was suggesting that we uninline all other instances where process.argv was inlined by this PR, not just this particular line @kylezs 🙂

**Comment by kylezs** [comment]
Told @j4m1ef0rd I'd take this over while he's away, but for now have deliberately left it to reduce bouncer changes that aren't directly addressing the flaky issues we've been seeing.

File: bouncer/shared/utils.ts (hunk 1 above)
Comment (j4m1ef0rd) [reply bouncer/shared/utils.ts:368]: Looks like the function is getting removed in #4958 anyway.

**Review by kylezs — APPROVED** [review]

File: bouncer/shared/utils.ts (hunk 1 above)
Comment (kylezs) [reply bouncer/shared/utils.ts:370]: I'll deal with it in that PR

**Comment by kylezs** [comment]
It looks like this PR introduces some new flakiness, I've only seen the connection issues on this PR. Should investigate this before merging.

**Comment by j4m1ef0rd** [comment]
Seems like the problem is with the connection to the BTC node. The error is happening when the swapping.ts test first tries to connect to the BTC node, no connection?. The false timeout message lead me astray for a while.
What could be causing this? @ahasna
I don't see how anything i did in this PR would affect that.

File: bouncer/commands/upgrade_network.ts [hunk 4]
```diff
@@ -116,14 +116,7 @@ async function main(): Promise<void> {
...
 }
 
 // Quite a long timeout, as the sequence of try-runtime runs takes some time.
-runWithTimeout(main(), 30 * 60 * 1000).catch((error) => {
-  console.error('upgrade_network exiting due to error: ', error);
-  if (process.exitCode === 0) {
-    process.exitCode = -1;
-  }
-});
+await executeWithTimeout(main(), 30 * 60 * 1000);
```
Comment (kylezs) [inline bouncer/commands/upgrade_network.ts:122]: Isn't this off by a factor of 1000 now?

**Comment by kylezs** [comment]
> Seems like the problem is with the connection to the BTC node.

I would first try rebasing not just to resolve conflicts but there have been a couple of changes to localnets recently

File: bouncer/full_bouncer.sh [hunk 5]
```diff
@@ -9,6 +9,7 @@ echo "Running nightly tests 🧪"
 ./tests/rotates_through_btc_swap.ts
 ./tests/btc_utxo_consolidation.ts
 ./tests/rotation_barrier.ts
+./tests/multiple_members_governance.ts
```
[the remark this answers is not in the document]
Comment (j4m1ef0rd) [reply bouncer/full_bouncer.sh:12]: I thought it might of been interfering with the boost test, but now i think it was fine. But it still makes sense to me, A test that changes the environment like that should not run at the same time as other tests, so i moved it to full bouncer instead.

File: bouncer/tests/all_concurrent_tests.ts [hunk 6]
```diff
@@ -65,7 +63,6 @@ runWithTimeout(runAllConcurrentTests(), 2000000)
     process.exit(0);
   })
   .catch((error) => {
-    console.error!('All concurrent tests timed out. Exiting.');
```
[the remark this answers is not in the document]
Comment (j4m1ef0rd) [reply bouncer/tests/all_concurrent_tests.ts:68]: This error message was logging whenever an error happened, not just a timeout, so it was very misleading. We already print a timeout log message and print the error here otherwise, so this error message was redundant.

File: bouncer/shared/utils.ts [hunk 7]
```diff
@@ -216,6 +216,26 @@ export const runWithTimeout = async <T>(promise: Promise<T>, millis: number): Pr
...
+export async function executeWithTimeout<T>(promise: Promise<T>, seconds: number): Promise<void> {
+  const start = Date.now();
+  await runWithTimeout(promise, seconds * 1000).catch((error) => {
+    console.error(error);
+    process.exit(-1);
+  });
+  const executionTime = (Date.now() - start) / 1000;
+
+  if (executionTime > seconds * 0.9) {
+    console.log(
```
Comment (kylezs) [inline bouncer/shared/utils.ts:229]: console.warn ?

**Review by kylezs — APPROVED** [review]
